lib/dpl/provider/pages.rb

```diff
@@ -28,7 +30,14 @@ def initialize(context, options)
 
         @gh_fqdn = fqdn
         @gh_url = options[:github_url] || 'github.com'
-        @gh_token = option(:github_token)
+        @gh_token = options[:github_token]
+        @gh_deploy_key = options[:deploy_key]
+        unless @gh_token || @gh_deploy_key
```

Comment:
This must go to the `check_auth` method

Reply:
Done in 7f4184d4f19bb2191106f05cbf260a95b78fece3.